### ISSUE TITLE
Modify softmax_cross_entropy to accept a meaningful value for normalize argument

### ISSUE THREAD

**Issue by gwtnb** [chainer/chainer#1565]
Current `normalize` argument of `softmax_cross_entropy` accepts a boolean value, but [its interpretation](https://github.com/pfnet/chainer/blob/v1.14.0/chainer/functions/loss/softmax_cross_entropy.py#L60-L64) is difficult for humans. We should use more meaningful values instead.

Ref: #1538

**Comment by gwtnb** [comment]
Of course, we have to fix the docstring.

**Comment by delta2323** [comment]
> We should use more meaningful values instead.

Could you explain here in more detail?

**Comment by gwtnb** [comment]
`softmax_cross_entropy` always perform normalization regardless of the value of `normalize`. `normalize` controls how the normalization is performed. This difficult interpretation of `normalize` is unreadable. We should modify `softmax_cross_entropy` to accept a meaningful value like `normalize='some_meaningful_string'`.

**Comment by bkvogel** [comment]
I would like to write a PR to fix this issue but I think I first need to understand the motivation for the behavior of `normalize` before we can decide on a name for a more meaningful value to the `normalize` argument.

I looked at the existing code and when `normalize=True` is used it looks like the losses are averaged across all of observations in a minibatch, exluding the observations in which `t` has a -1 label. This seems reasonable. I would have guessed that `normalize=False` would do the same thing without taking the average, so that all losses would be summed exluding the observations with -1 labels. However, it looks like the code is doing something different: if `normalize=False` is used, the summ of the losses for all observations without -1 labels is divided by the mini-batch size, right? I do not understand the motivation for this behavior when `normalize=False` is used. Do other frameworks do something similar?

**Comment by delta2323** [comment]
We are discussing the problem in the other context, or loss functions with finer outputs. We decided to  make loss functions (at least `SigmoidCrossEntropy` and `SoftmaxCrossEntropy`) output loss values without reducing (i.e. do not take summation nor average along axes) and remove `normalize` option from these functions. We can reproduce the original behavior of `normalize` option by applying `F.Average` or `sum` to the output of loss functions. We expect additional overhead is moderate.
